internal/dag/dag.go

```diff
@@ -258,6 +262,34 @@ type LocalRateLimitPolicy struct {
 	ResponseHeadersToAdd map[string]string
 }
 
+// HeaderHashOptions contains options for hashing a HTTP header.
+type HeaderHashOptions struct {
+	// HeaderName is the name of the header to hash.
+	HeaderName string
+}
+
+// RequestHashPolicy holds configuration for calculating hashes on
+// an individual request attribute.
+type RequestHashPolicy struct {
+	// Terminal determines if the request attribute is present, hash
+	// calculation should stop with this element.
+	Terminal bool
+
+	// HeaderHashOptions is set when a header hash is desired.
+	HeaderHashOptions *HeaderHashOptions
+}
+
+// LoadBalancerHashPolicy holds load balancing hash policies for load
+// balancing based on request attributes.
+type LoadBalancerHashPolicy struct {
+	// Strategy is the Contour load balancing strategy chosen.
+	Strategy string
```

Comment:
I was thinking of a refactor actually where the `Cookie` lb policy would generate request hash policies at the `dag` level instead of the lower level translation to Envoy config

this would probably make the `Strategy` field here obsolete since the route doesn't need to know which type of load balancing strategy is in use, just if it has some hash policies to set